apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/page.tsx

```diff
@@ -13,7 +15,13 @@ export default async function DocumentPage({
   params: { projectId: string; commitUuid: string; documentUuid: string }
   searchParams: QueryParams
 }) {
-  const { workspace } = await getCurrentUser()
+  const session = await getSafeCurrentUser()
+
+  if (!session) {
+    redirect(ROUTES.auth.login)
+  }
```

Comment:
This fix the issue. The thing is we want to do this in all authenticated pages. What would be the best abstraction. 

I'm trying HOC to get the session or redirect but with async components typescript is complaining. 

What other ideas we could use? Or is fine doing this same code in all the authenticated pages?

Reply:
i think the appropiate way for doing this is middlewares

Reply:
The thing is that we have this check on private layout
https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/app/(private)/layout.tsx#L25-L28

And also this on fetcher 
https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/hooks/useFetcher.ts#L35-L42

Not sure we need to do a check with error by page inside `(private)` routes.